src/java_bytecode/java_bytecode_language.cpp

```diff
@@ -408,6 +409,30 @@ bool java_bytecode_languaget::convert_single_method(
       string_preprocess.code_for_function(symbol, symbol_table);
     INVARIANT(
       generated_code.is_not_nil(), "Couldn't retrieve code for string method");
+    // String solver can make calls to functions that haven't yet been seen.
+    // Add these to the needed_lazy_methods collection
+    if(needed_lazy_methods)
+    {
+      for(const_depth_iteratort it = generated_code.depth_cbegin();
+          it != generated_code.depth_cend();
+          ++it)
```

Comment:
can you use `std::for_each` instead

Reply:
Not sure this would add any clarity in this case.